### PR TITLE
feat: Extract event listeners functions for Deposit and ProposalExecution to EVM folder from Sygma class

### PR DESCRIPTION
**Pull request by enemycnt**
This PR is based on #204 and shoud be merged after it
## Related Issue Or Context

Closes: #193 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: packages/sdk/.eslintrc.cjs [hunk 1]
```diff
@@ -1,3 +1,12 @@
 module.exports = {
   extends: '../../.eslintrc.cjs',
+  parserOptions: {
+    ecmaFeatures: {
+      jsx: true,
+    },
+    // ecmaVersion: 12,
```
Comment (wainola) [inline packages/sdk/.eslintrc.cjs:7]: I was experiencing issues with the linter by myself in my last PR. I have used this settings and works pretty fine. Why the commented code? If we are not using it, we should remove it

File: packages/sdk/src/chains/EVM/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+ * @param provider - JsonRpcProvider | Web3Provider
+ * @returns {Promise<BigNumber | boolean>}
+ */
+export async function isEIP1559MaxFeePerGas(provider: providers.Provider): Promise<BigNumber> {
+  try {
+    const feeData = await provider.getFeeData();
+    const { gasPrice } = feeData;
+    return gasPrice as BigNumber;
+  } catch (error) {
+    console.error('error getting EIP 1559');
```
Comment (wainola) [inline packages/sdk/src/chains/EVM/helpers.ts:195]: can we log the error anyways? just a smal convenience

File: packages/sdk/src/chains/EVM/utils/approvesAndChecksFns.ts
Comment (wainola) [inline packages/sdk/src/chains/EVM/utils/approvesAndChecksFns.ts:1]: I would suggest here, do the same that we have in other files, that's it changing all the logs that are in the `catch` side to be `console.error`

**Review by wainola — CHANGES_REQUESTED** [review]

File: packages/sdk/.eslintrc.cjs (hunk 1 above)
Comment (enemycnt) [reply packages/sdk/.eslintrc.cjs:7]: It will be rebased against deposit's PRs

File: packages/sdk/src/chains/EVM/utils/approvesAndChecksFns.ts
Comment (enemycnt) [reply packages/sdk/src/chains/EVM/utils/approvesAndChecksFns.ts:1]: Done

**Review by wainola — DISMISSED** [review]

File: packages/sdk/src/chains/EVM/helpers.ts [hunk 3]
```diff
@@ -26,19 +28,22 @@ export const addPadding = (covertThis: string | number, padding: number): string
...
- * @description creates a resource id based on the contract address
+ * Creates a resource id based on the contract address
+ *
  * @param contractAddress
  * @param domainID
  * @returns {string}
  */
-export const createResourceID = (contractAddress: string, domainID: number): string => {
-  return toHex(contractAddress + toHex(domainID, 1).substr(2), 32);
+export const createResourceID = (contractAddress: string, domainID: string | number): string => {
```
Comment (MakMuftic) [inline packages/sdk/src/chains/EVM/helpers.ts:37]: I don't understand the use case for this method, as we are not defining resource ID based on some logic like this.

**Review by MakMuftic — CHANGES_REQUESTED** [review]

File: packages/sdk/src/chains/EVM/helpers.ts (hunk 3 above)
Comment (enemycnt) [reply packages/sdk/src/chains/EVM/helpers.ts]: it was copied from solidity helpers. I've removed it as unused

**Review by MakMuftic — APPROVED** [review]

**Review by wainola — APPROVED** [review]

**Pull request merged**